drivers/isl29125/isl29125.c

```diff
@@ -0,0 +1,272 @@
+    conf1 |= ISL29125_CON1_SYNCOFF; /* TODO: implement SYNC mode configuration */
+
+    return conf1;
+}
+
+/***********************************************************************
+ * public API implementation
+ **********************************************************************/
+
+int isl29125_init(isl29125_t *dev, i2c_t i2c, uint8_t address,
+        isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution)
+{
+    int ret;
+
+    DEBUG("isl29125_init\n");
+
+    /* initialize device descriptor */
+    dev->i2c = i2c;
+    dev->address = address;
+    dev->res = resolution;
+    dev->range = range;
+
+    /* configuration 1: operation mode, range, resolution */
+    char conf1 = parse_conf1(mode, range, resolution);
+
+    /* TODO: implement configuration 2: infrared compensation configuration */
+
+    /* TODO: implement configuration 3: interrupt mode configuration */
+
+    /* acquire exclusive access to the bus */
+    DEBUG("isl29125_init: i2c_acquire\n");
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    DEBUG("isl29125_init: i2c_init_master\n");
+    i2c_init_master(i2c, I2C_SPEED_NORMAL);
+
+    /* verify the device ID */
+    DEBUG("isl29125_init: i2c_read_reg\n");
+    char reg_id;
+    ret = i2c_read_reg(dev->i2c, dev->address, ISL29125_REG_ID, &reg_id);
+    if ((reg_id == ISL29125_ID) && (ret == 1)) {
+        DEBUG("isl29125_init: ID successfully verified\n");
+    }
+    else {
+        DEBUG("isl29125_init: ID could not be verified, ret: %i\n", ret);
+        ret = -1;
+        goto release;
+    }
+
+    /* configure and enable the sensor */
+    DEBUG("isl29125_init: i2c_write_reg(ISL29125_REG_RESET)\n");
+    ret = i2c_write_reg(dev->i2c, address, ISL29125_REG_RESET, ISL29125_CMD_RESET);
+    if (ret != 1) {
+        DEBUG("isl29125_init: i2c_write_reg(ISL29125_REG_RESET) failed: %i\n", ret);
+        ret = -1;
+        goto release;
+    }
+    DEBUG("isl29125_init: i2c_write_reg(ISL29125_REG_CONF1)\n");
+    ret = i2c_write_reg(dev->i2c, address, ISL29125_REG_CONF1, conf1);
+    if (ret != 1) {
+        DEBUG("isl29125_init: i2c_write_reg(ISL29125_REG_CONF1) failed: %i\n", ret);
+        ret = -1;
+        goto release;
+    }
+
+release:
+    /* release the I2C bus */
+    DEBUG("isl29125_init: i2c_release\n");
+    i2c_release(dev->i2c);
+    if (ret < 1) {
+        DEBUG("isl29125_init: fail\n");
+        return -1;
+    }
+
+    DEBUG("isl29125_init: success\n");
+    return 0;
+}
+
+int isl29125_read_rgb(isl29125_t *dev, isl29125_rgb_t *dest)
+{
+    /* acquire exclusive access to the bus */
+    i2c_acquire(dev->i2c);
+
+    /* read values */
+    char bytes[6];
+    int ret = i2c_read_regs(dev->i2c, dev->address, ISL29125_REG_GDLB, bytes, 6);
+
+    /* release the I2C bus */
+    i2c_release(dev->i2c);
+    if (ret < 6) {
+        return -1;
+    }
+
+    /* convert readings */
+
+    int resfactor = (dev->res == ISL29125_RESOLUTION_12) ? 4 : 0;
```

Comment:
Don't know if this would result in  too much docu but I wondered where the `4` comes from